src/deploy.ts

```diff
@@ -0,0 +1,35 @@
+import {resolve} from 'path';
+
+export default async function deploy({rootPath, sourceDirectory, domain, pr, surgeService, fileService, githubService}) {
+  const surgeDomain = `https://${domain}${pr ? `-pr-${pr}` : ''}.surge.sh/`;
+
+  const updateStatus = await createStatusUpdater({githubService, pr, domain});
+  updateStatus('pending', 'Deploying to surge');
+
+  try {
+    const projectPath = `${rootPath}/${sourceDirectory}`;
+    if (!fileService.exists(projectPath)) {
+      throw(new Error(`${projectPath} does not exist`));
+    }
+    await surgeService(['publish', '--project', projectPath, '--domain', surgeDomain]);
+    updateStatus('success', 'Deployed to surge.sh', surgeDomain);
+  } catch (e) {
+    updateStatus('error', 'Deployment failed');
+    throw e;
+  }
+}
+
+async function createStatusUpdater({githubService, pr, domain}) {
+  if (!githubService || !pr) {
```

Comment:
I guess it's rather a matter of style, but isn't it better API-wise to always assume the github service exists?